social_core/backends/jira.py

```diff
@@ -0,0 +1,102 @@
+"""
+Atlassian Jira Server OAuth backend, docs at:
+    https://developer.atlassian.com/cloud/jira/platform/jira-rest-api-oauth-authentication/
+"""
+import six
+
+from oauthlib.oauth1 import SIGNATURE_RSA, SIGNATURE_TYPE_QUERY, SIGNATURE_TYPE_AUTH_HEADER
+from requests_oauthlib import OAuth1
+from .oauth import BaseOAuth1
+from ..exceptions import AuthTokenError
+
+
+class JiraOAuth(BaseOAuth1):
+    """Jira OAuth1 Authentication backend"""
+    name = 'jira'
+    ID_KEY = 'accountId'
+    _ACCESS_TOKEN_URL = '{scheme}://{host}/jira/plugins/servlet/oauth/access-token'
+    _AUTHORIZATION_URL = '{scheme}://{host}/jira/plugins/servlet/oauth/authorize'
+    _REQUEST_TOKEN_URL = '{scheme}://{host}/jira/plugins/servlet/oauth/request-token'
+    _USER_DETAILS_URL = '{scheme}://{host}/rest/api/2/myself'
+    ACCESS_TOKEN_METHOD = 'POST'
+
+    def auth_url(self):
+        token = self.get_unauthorized_token()
+        return self.AUTHORIZATION_URL + '?oauth_token=' + token.get('oauth_token')
+
+    def get_unauthorized_token(self):
+        return self.get_json(
+            self.REQUEST_TOKEN_URL,
+            auth=self.oauth_auth()
+        )
+
+    @property
+    def ACCESS_TOKEN_URL(self):
+        return self._ACCESS_TOKEN_URL.format(**self.jira_url_params)
+
+    @property
+    def AUTHORIZATION_URL(self):
+        return self._AUTHORIZATION_URL.format(**self.jira_url_params)
```

Comment:
There's no need to define this property, you can define the method `authorization_url` instead.